### ISSUE TITLE
[BUG] - Propagation delay increase with 1.29.0+ with start of rewards calculation (at the 48 hour mark) for about 24 hours

### ISSUE THREAD

**Issue by Straightpool** [input-output-hk/cardano-node#3167]
**Internal/External**
*External*   otherwise.

**Area**
*Other*  Block propagation

**Summary**
After the 48 hour mark (validated in both epoch 288 and 289) with the start of the rewards calculation block propagation delays increase significantly for little less than 24 hours (until end of rewards calculation).

**Steps to reproduce**
1. Start block producer setup on 1.29.0 on mainnet
2. Wait on epoch switch + 48 hours 
3. Wait on minted block within about 24 hours
4. Track propagation delays
5. See error (especially when you compare with propagation delays on 1.29.0 before Thu 04 Sep 2021 (21:45:00 UTC)
6. After about 24 hours the propagation delays return back to normal 

**Expected behavior**
Propagation delays should not increase with 1.29.0 with or without a parallel rewards calculation on the node.

**System info (please complete the following information):**
- OS Name: Ubunto
- OS Version: 20.04.1 LTS
- Node version:
 `cardano-node 1.29.0 - linux-x86_64 - ghc-8.10
git rev 4c59442958072657812c6c0bb8e0b4ab85ce1ba2`
- CLI version:
 `cardano-cli 1.29.0 - linux-x86_64 - ghc-8.10
git rev 4c59442958072657812c6c0bb8e0b4ab85ce1ba2`

**Screenshots and attachments**
_Block propagation before the 48 hour mark on 1.29.0_
![image](https://user-images.githubusercontent.com/42584250/132128032-6c3dba67-14ac-4ac4-8b0d-65c4cb8c7f01.png)

_Block producer log:_
{"thread":"111","loc":null,"data":{"val":{"kind":"TraceForgedBlock","blockPrev":"6e3aeabd954f9399a926768a1344fa49914bff15fc86220d7bb1a699200cf98e","slot":39201349,"block":"96e80c3c7b2b2a2c95136cffda062212dc1d96b91be578fb60e1128ab74ff2a4","blockNo":6201140},"credentials":"Cardano"},"sev":"Info","env":"1.29.0:4c594","msg":"","app":[],"host":"ub20cn","pid":"156659","ns":["cardano.node.Forge"],"at":"2021-09-04T15:00:40.02Z"}
{"thread":"101","loc":null,"data":{"newtip":"96e80c3c7b2b2a2c95136cffda062212dc1d96b91be578fb60e1128ab74ff2a4@39201349","chainLengthDelta":1,"kind":"TraceAddBlockEvent.AddedToCurrentChain"},"sev":"Notice","env":"1.29.0:4c594","msg":"","app":[],"host":"ub20cn","pid":"156659","ns":["cardano.node.ChainDB"],"at":"2021-09-04T15:00:40.04Z"}
is a gap of 0.02S

_Relay log:_
{"thread":"48667","loc":null,"data":{"kind":"ChainSyncClientEvent.TraceDownloadedHeader","peer":[....]{"remote":},"slot":39201349,"block":"96e80c3c7b2b2a2c95136cffda062212dc1d96b91be578fb60e1128ab74ff2a4","blockNo":6201140},"sev":"Info","env":"1.29.0:4c594","msg":"","app":[],"host":"ub20r1","pid":"135872","ns":["cardano.node.ChainSyncClient"],"at":"2021-09-04T15:00:40.04Z"}
to first
{"thread":"609","loc":null,"data":{"kind":"TraceBlockFetchServerSendBlock","block":"96e80c3c7b2b2a2c95136cffda062212dc1d96b91be578fb60e1128ab74ff2a4"},"sev":"Info","env":"1.29.0:4c594","msg":"","app":[],"host":"ub20r1","pid":"135872","ns":["cardano.node.BlockFetchServer"],"at":"2021-09-04T15:00:40.10Z"}
is a gap of 0.06 seconds.

_Block propagation time after the 48 hour mark on 1.29.0_
![image](https://user-images.githubusercontent.com/42584250/132128013-2cac9555-19c7-40ea-af3c-07d575874216.png)

_Block producer log:_
{"thread":"111","loc":null,"data":{"val":{"kind":"TraceForgedBlock","blockPrev":"0b0c9df3c3313197e760bafc7a9370e7be676effc7aeecde7231b4e07898c2b2","slot":39271659,"block":"26541add2818066ae41b07f8788f646bb4954202057f83841bc4a3e1a7ab8024","blockNo":6204568},"credentials":"Cardano"},"sev":"Info","env":"1.29.0:4c594","msg":"","app":[],"host":"ub20cn","pid":"156659","ns":["cardano.node.Forge"],"at":"2021-09-05T10:32:30.15Z"}
{"thread":"101","loc":null,"data":{"newtip":"26541add2818066ae41b07f8788f646bb4954202057f83841bc4a3e1a7ab8024@39271659","chainLengthDelta":1,"kind":"TraceAddBlockEvent.AddedToCurrentChain"},"sev":"Notice","env":"1.29.0:4c594","msg":"","app":[],"host":"ub20cn","pid":"156659","ns":["cardano.node.ChainDB"],"at":"2021-09-05T10:32:30.29Z"}
is a gap of 0.14S.

**This is an increase of factor 7 on the block producer.**

_Relay log:_
{"thread":"48667","loc":null,"data":{"kind":"ChainSyncClientEvent.TraceDownloadedHeader","peer":{"remote":[...],"slot":39271659,"block":"26541add2818066ae41b07f8788f646bb4954202057f83841bc4a3e1a7ab8024","blockNo":6204568},"sev":"Info","env":"1.29.0:4c594","msg":"","app":[],"host":"ub20r1","pid":"135872","ns":["cardano.node.ChainSyncClient"],"at":"2021-09-05T10:32:30.37Z"}
to first
{"thread":"609","loc":null,"data":{"kind":"TraceBlockFetchServerSendBlock","block":"26541add2818066ae41b07f8788f646bb4954202057f83841bc4a3e1a7ab8024"},"sev":"Info","env":"1.29.0:4c594","msg":"","app":[],"host":"ub20r1","pid":"135872","ns":["cardano.node.BlockFetchServer"],"at":"2021-09-05T10:32:30.68Z"}
is a gap of 0.31seconds. 

**This is an increase of factor 5.1 on the relay.**

_After less than 24 hours the propagation delay is back to normal:_
![image](https://user-images.githubusercontent.com/42584250/132140088-cfcf3cc7-b061-4049-bd38-9f43f928b985.png)

**Additional context**
- Node restart does not fix the issue
- This is not a reporting issue, other setups tracking propagation are showing the same symptoms
- Throwing additional CPU cores at the nodes might help for an individual hop, this is clearly connected to rewards calculation which kicks in 48 hours after epoch start, most likely a CPU scheduling issue
- This issue is related to https://github.com/input-output-hk/ouroboros-network/issues/2913 as it can lead to block collision

**Comment by gufmar** [comment]
Same behaviour reported by many different SPOs with exact same pattern

**Comment by rphair** [comment]
Has anyone tried running with the newer node versions (none of them yet tagged as releases) up through `1.30.1` to see if this problem is mitigated?

I can also confirm any block we've missed not attributable to some other factor has been in the 48 to 72 hour range in each epoch, and that it does appear to compound with https://github.com/input-output-hk/ouroboros-network/issues/2913.

**Comment by weebl2000** [comment]
Let's see if the PR mentioned solves this. If not, I might have a suggestions - but I'll test it out first.

**Comment by reqlez** [comment]
Observed an interesting behavior ( second block was ghosted today ). The only 2 blocks that were ever ghosted at my pool, both were due to the same operator's  propagation issues.

https://adapools.org/groups/fimi-146

First it was FIMI, and now it's FIDA And it seems it is the same operator.

Now, I don't know if this has to do with the location/latency ( Vietnam ) or the operator. But an interesting find. I feel this might not be a coincidence.

I did see some code was merged, let's hope this fixes the issues. i'm assuming this is coming out in 1.32.0 ?
